bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -106,7 +106,6 @@ void ClientsManager::loadInfoFromReservedPages() {
       auto replyHeader = (ClientReplyMsgHeader*)(scratchPage_.data());
       auto replyPtr = std::make_shared<ClientReplyMsg>(myId_, replyHeader->replyLength);
       ConcordAssert(replyHeader->msgType == 0 || replyHeader->msgType == MsgCode::ClientReply);
-      ConcordAssert(replyHeader->currentPrimaryId == 0);
```

Comment:
I agree, this assertion does not make sense, the Client should track the current Primary after initially broadcasting request to all Replicas. Then in the responses it should be able to get the current Primary and send to it the following requests, taking in consideration the timeout for broadcasting again if necessary.